charforge-gui/backend/app/core/middleware.py

```
@@ -0,0 +1,106 @@
+        db.refresh(default_user)
+        return default_user
+    except IntegrityError:
+        # Another request created the user, rollback and fetch it
+        db.rollback()
+        default_user = db.query(User).filter(User.id == settings.DEFAULT_USER_ID).first()
+        if not default_user:
+            # This should not happen, but handle it gracefully
+            raise HTTPException(
+                status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+                detail="Failed to create or retrieve default user"
+            )
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)